Reduce references to source pathnames down to the respective basenames so
that builds are reproducible between workspaces.
The patch is needed to avoid embedding path information from the build area
in the compiled binares (in particular in error messages).

--- rsync-3.5.0/t_stub.c.orig	2026-08-02 21:35:03.000000000 +0200
+++ rsync-3.5.0/t_stub.c	2026-08-13 12:09:09.291529764 +0200
@@ -20,6 +20,7 @@
  */
 
 #include "rsync.h"
+#include <libgen.h>
 
 int do_fsync = 0;
 int inplace = 0;
@@ -74,7 +75,7 @@
  void _exit_cleanup(int code, const char *file, int line)
 {
 	fprintf(stderr, "exit(%d): %s(%d)\n",
-		code, file, line);
+		code, basename((char *)file), line);
 	exit(code);
 }
 
--- rsync-3.5.0/log.c.orig	2026-07-20 06:07:05.000000000 +0200
+++ rsync-3.5.0/log.c	2026-08-13 12:13:25.081670218 +0200
@@ -23,6 +23,7 @@
 #include "itypes.h"
 #include "inums.h"
 #include "rounding.h"	/* EXTRA_ROUNDING, so log_delete() aligns its file_struct */
+#include <libgen.h>
 
 extern int dry_run;
 extern int am_daemon;
--- rsync-3.5.0/cleanup.c.orig	2026-05-24 04:31:52.000000000 +0200
+++ rsync-3.5.0/cleanup.c	2026-08-13 12:14:01.508140334 +0200
@@ -21,6 +21,7 @@
  */
 
 #include "rsync.h"
+#include <libgen.h>
 
 extern int dry_run;
 extern int am_server;
